rtps-r52/tests/mailbox.c

```diff
@@ -26,3 +32,51 @@ int test_mbox_lsio_loopback(void)
     test_end("test_mbox_lsio_loopback", rc);
     return rc;
 }
+
+int test_mbox_rtps_hpps(int mbox_in, int mbox_out){
+    enum hpsc_mbox_test_rc rc = HPSC_MBOX_TEST_SUCCESS;
+    struct hpsc_mbox *mbox_hpps = dev_get_mbox(DEV_ID_MBOX_HPPS_RTPS); //DEV_ID_MBOX_HPPS_TRCH
+    //struct hpsc_mbox *mbox_hpps = dev_get_mbox(DEV_ID_MBOX_LSIO);
+    assert(mbox_hpps);
+
+    test_begin("test_mbox_rtps_hpps");	
+
+    const char* name = "TEST_LINK";
+    struct link *test_link = link_mbox_connect(name, mbox_hpps,mbox_in ,mbox_out , 0, 1); //claim a pair of mailboxes as client. one from channel 3 to 4 and one from channel 4 to 3
+    assert(link);
+
+    int bytes; //bytes read 
+
+    char write_buf [HPSC_MSG_SIZE];
+    char read_buf [HPSC_MSG_SIZE];
+    char payload [HPSC_MSG_PAYLOAD_SIZE];
+
+    memset(&payload, 0, sizeof(payload));
+    hpsc_msg_ping(write_buf, HPSC_MSG_SIZE, payload, sizeof(payload));
+
+
+    bytes = link_request(test_link, 
+                     SEND_RCV_TIME_OUT, write_buf, sizeof(write_buf),
+                     SEND_RCV_TIME_OUT, read_buf, sizeof(read_buf),
+                     RTEMS_EVENT_2);
+
+    //send failure/timeout
+    if (bytes == -1) {
+        rc = HPSC_MBOX_TEST_CHAN_WRITE;
+        link_disconnect(test_link);
+        test_end("test_mbox_rtps_hpps",rc);
+        return rc;
+    }
+    //receive timeout
+    else if (bytes == -2) {
+        rc = HPSC_MBOX_TEST_CHAN_NO_RECV;
+        link_disconnect(test_link);
+        test_end("test_mbox_rtps_hpps",rc);
+        return rc;
+    }
+
+    printf("%s\n", read_buf); //remove later
```

Comment:
Remove now?  Verify in code that you got the expected response?  See `lib/hpsc-test/link.c`.

Also, maybe cleanup the comments earlier in this function.